codex/utils/asyncstatemachine.nim

```diff
@@ -84,18 +90,20 @@ proc start*(machine: Machine, initialState: State) =
 
   if machine.scheduled.isNil:
     machine.scheduled = newAsyncQueue[Event]()
-  machine.scheduling = machine.scheduler()
+
   machine.started = true
+  machine.scheduler()
+    .track(machine)
+    .catch((err: ref CatchableError) =>
+      error("Error in scheduler", error = err.msg)
```

Comment:
Good that you make the error handling explicit here. The scheduler should never fail. If it does, then it's a programmer error. So ideally I'd like the program to crash, not hide the error in the logs.

Reply:
Oops, I forgot to address this. Will open a new PR.